minet/instagram/api_scraper.py

```diff
@@ -49,41 +138,196 @@ def __init__(self, cookie="firefox"):
             raise InstagramInvalidCookieError
 
         self.cookie = cookie
+        self.magic_token = None
+        self.nb_calls = 0
+        self.nb_calls_2900 = 0
+        self.retryer = create_request_retryer(
+            min=INSTAGRAM_MIN_TIME_RETRYER,
+            additional_exceptions=[InstagramTooManyRequestsError, InstagramError500],
+        )
 
-    def request_json(self, url):
+    @retrying_method()
+    def request_json(self, url, magic_token=False):
+        headers = {"Cookie": self.cookie}
+        if magic_token:
+            headers["X-IG-App-ID"] = self.magic_token
         err, response, data = request_json(
-            url, pool=self.pool, spoof_ua=True, headers={"Cookie": self.cookie}
+            url,
+            pool=self.pool,
+            spoof_ua=True,
+            headers=headers,
         )
 
         if err:
             raise err
 
+        if response.status == 429:
+            raise InstagramTooManyRequestsError
+
+        if response.status == 500:
+            raise InstagramError500
+
         if response.status >= 400:
+            print(response.status, response.data)
             raise InstagramPublicAPIInvalidResponseError
 
+        if self.nb_calls == 10:
```

Comment:
It would be better to frame/name those constants as something like "little" wait and "big" wait or something like this but keep the opportunity to change them to different numbers.